### PR TITLE
[Workspace] Remove temporary JSON transformers

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
Depends on #766

**Comment by ddunbar** [comment]
IMHO, it would be better to just write this using a `Version.init(json:)` function in an extension... I don't think the code savings justify the new API.

**Comment by aciidb0mb3r** [comment]
@ddunbar done

File: Sources/Commands/misc.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import Basic
+import Utility
+import SourceControl
+
+extension Version {
+    init?(json: JSON) {
+        switch json {
+        case .string(let str):
+            self.init(str)
```
Comment (ddunbar) [inline Sources/Commands/misc.swift]: FWIW, you could just use:
```
guard case let .string(str) else { return nil }
self.init(str)
```

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Pull request merged**